lib/kitchen/note_element.rb

```diff
@@ -15,6 +15,7 @@ def initialize(node:, document: nil)
             document: document,
             enumerator_class: NoteElementEnumerator,
             short_type: :note)
+      @detected_note_title_key = detected_note_title_key
```

Comment:
no need for this as `@detected_note_title_key` is only used from within `detected_note_title_key` when it is set.